lib/sqeduler/service.rb

```diff
@@ -102,7 +102,8 @@ def config_redis_pool
         redis_pool = if config.redis_pool
           config.redis_pool
         else
-          redis = { :namespace => "sqeduler" }.merge(config.redis_hash)
+          redis = config.redis_hash.dup
+          redis[:namespace] = "sqeduler" if redis[:namespace].nil?
```

Comment:
My code doesn't really change the behavior. It changes what gets passed in to RedisConnection.

That is, if config.redis_hash is of type `ActiveSupport::HashWithIndifferentAccess`, the same type gets passed in and not `Hash`.

This has implications since Redis requires the config hash to have symbols as keys. If `config.redis_hash` is of type `ActiveSupport::HashWithIndifferentAccess` but with string as keys, the original code converted the config hash into `Hash`, which would then have string as keys.

That said, I will rather go with what @alyssa has suggested below.